exatomic/adf/tape21.py

```diff
@@ -0,0 +1,362 @@
+        normal modes. With the equation,
+
+        .. math::
+            \\mu_i = \\left(\\sum_k^{3N} l_{CARTk,i}^2\\right)^{-1}
+
+        Note:
+            This assumes that the normal modes have already been placed in the
+            :code:`['dx', 'dy', 'dz']` columns.
+
+        Args:
+            data (:class:`pandas.DataFrame`): Data frame the has the non-mass-weighted
+                                              cartesian normal modes.
+            symbol (:obj:`list`): List-like object that has the atomic symbols.
+
+        Returns:
+            r_mass (:class:`numpy.ndarray`): Array containing the calculated reduced
+                                             masses.
+        '''
+        cols = ['dx', 'dy', 'dz']
+        # get the isotopic masses of the unique atoms
+        mapper = sym2isomass(symbol)
+        # get a list of the isotopic masses
+        mass = list(map(mapper.get, symbol))
+        mass = np.repeat(mass, 3).astype(float)
+        mass = mass.reshape(data[cols].shape)
+        disps = data[cols].values
+        norms = np.linalg.norm(disps*np.sqrt(mass))
+        norms = 1/norms
+        disps *= norms
+        r_mass = np.sum(np.square(disps))
+        r_mass = 1/r_mass
+        return r_mass
+
+    def _intme(self, fitem, idx=0):
+        return int(self[fitem[idx]+1].split()[0])
+
+    def _dfme(self, fitem, dim, idx=0):
+        start = fitem[idx] + 2
+        col = min(len(self[start].split()), dim)
+        stop = np.ceil(start + dim / col).astype(np.int64)
+        return self.pandas_dataframe(start, stop, col).stack().values
+
+    def parse_frequency(self, cart=True):
+        '''
+        ADF frequency parser.
+
+        Note:
+            This will toss a warning if it cannot find the mass-weighted normal modes
+            which must be used to generate the displaced structures for vibrational
+            averaging. Also, it will be unable to calculate the reduced masses as it will
+            have normalized cartesian coordinates where it expects normalized
+            mass-weighted cartesian normal modes.
+
+        Args:
+            cart (:obj:`bool`, optional): Parse the normalized cartesian coordinates or
+                                          the mass-weighted normal modes. Defaults to
+                                          :code:`True`.
+        '''
+        # search flags
+        _renorm = "NormalModes_RAW"
+        _recartnorm = "Normalmodes"
+        _refreq = "Frequencies"
+        _refreqexc = r"\bFrequencies\b"
+        _rekey = r"\bFreq\b"
+        found = self.find(_refreq, _renorm, _recartnorm, keys_only=True)
+        key = self.regex(_rekey, _refreqexc, keys_only=True)
+        # need to do this to ensure that we only match the data in the Freq block
+        found_freq = []
+        for k in key[_rekey]:
+            for f in found[_refreq]:
+                if f-1 == k:
+                    found_freq.append(f)
+                    break
```

Comment:
Is this intended to break out of both for loops (as written it breaks out of the innermost only I think)?